### PR TITLE
Friends App

### PR DESCRIPTION
**Pull request by kovalov**
### Friends App

[Demo](https://kovalov.github.io/kottans-friends-app/) |
[Code base](https://github.com/kovalov/kottans-friends-app)

The code is submitted in a dedicated feature branch.

Only code files are submitted.

Please, review.

### CODE REVIEW

**Comment by zonzujiro** [comment]
image

And how you sorting the same data by name and by age? 🤔

**Comment by zonzujiro** [comment]
image

Looks like age sorting doesn't work.

Also, you should render age on cards :)

**Comment by zonzujiro** [comment]
Also, be sure that you don't have [common mistakes](https://kottans.org/documentation/docs/doc/code-review#friends-app)

**Comment by kovalov** [comment]
> image
> 
> Looks like age sorting doesn't work.
> 
> Also, you should render age on cards :)
> 

Age is showing near the full name. Should I implement it another way?

**Comment by kovalov** [comment]
Have rewrite the code according to common mistakes list. Please check.

**Review by zonzujiro — APPROVED** [review]
LGTM

File: submissions/kovalov/friends-app/app.js [hunk 1]
```diff
@@ -0,0 +1,176 @@
+const URL =
+  'https://randomuser.me/api/?results=20&nat=us,dk,fr,gb&inc=gender,name,email,dob,phone,picture';
+
+const userListContainerElement = document.querySelector('.user-list');
+const formElement = document.querySelector('.form');
+const resetButtonElement = document.querySelector(
+  '.form__reset-button'
+);
+// const formSearchInputElement = document.querySelector(
```
Comment (zonzujiro) [inline submissions/kovalov/friends-app/app.js:9]: :) Leftover?

File: submissions/kovalov/friends-app/app.js [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+};
+
+const getFilteredUsersByGender = (selectedGender) => {
+  if (selectedGender === 'male' || selectedGender === 'female') {
+    sortedUserData = filterUsersByGender(selectedGender);
+  }
+};
+
+const getSortedUsersByAge = (value) => {
+  if (value === 'ageAscending' || value === 'ageDescending') {
```
Comment (zonzujiro) [inline submissions/kovalov/friends-app/app.js:143]: No need to call function if will do nothing.

Instead you can have object with keys from your values of inputs and values of object from sorters.

File: submissions/kovalov/friends-app/app.js [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+};
+
+const getSortedUsersByName = (value) => {
+  if (value === 'nameAscending' || value === 'nameDescending') {
+    nameSorters[value]();
+  }
+};
+
+const handleFormInputs = () => {
+  const { value: searchValue } = formElement.search;
```
Comment (zonzujiro) [inline submissions/kovalov/friends-app/app.js:155]: Looks like you're using values as names for inputs :)

**Review by zonzujiro — CHANGES_REQUESTED** [review]

File: submissions/kovalov/friends-app/app.js (hunk 3 above)
Comment (zonzujiro) [reply submissions/kovalov/friends-app/app.js:155]: Ah, no. All good

**Review by zonzujiro — APPROVED** [review]
Great job!

**Pull request merged**